cmd/kubeadm/app/constants/constants.go

```diff
@@ -446,36 +441,22 @@ func AddSelfHostedPrefix(componentName string) string {
 	return fmt.Sprintf("%s%s", SelfHostingPrefix, componentName)
 }
 
-// CreateTempDirForKubeadm is a function that creates a temporary directory under /etc/kubernetes/tmp (not using /tmp as that would potentially be dangerous)
+// CreateTempDirForKubeadm is a function that creates a temporary directory
 func CreateTempDirForKubeadm(dirName string) (string, error) {
-	tempDir := path.Join(KubernetesDir, TempDirForKubeadm)
-	// creates target folder if not already exists
-	if err := os.MkdirAll(tempDir, 0700); err != nil {
-		return "", errors.Wrapf(err, "failed to create directory %q", tempDir)
-	}
-
-	tempDir, err := ioutil.TempDir(tempDir, dirName)
+	tempDir, err := ioutil.TempDir("", dirName)
 	if err != nil {
 		return "", errors.Wrap(err, "couldn't create a temporary directory")
 	}
 	return tempDir, nil
 }
 
-// CreateTimestampDirForKubeadm is a function that creates a temporary directory under /etc/kubernetes/tmp formatted with the current date
+// CreateTimestampDirForKubeadm is a function that creates a temporary directory formatted with the current date
 func CreateTimestampDirForKubeadm(dirName string) (string, error) {
```

Comment:
This can impact upgrades though, as "backups" will be now placed in `/tmp` instead of `/etc/kubernetes/tmp/{kubeadm-backup-manifests,kubeadm-backup-etcd}-date`.

Reply:
`staticpods.CleanupDirs` should clean them up though, the only drawback I see with this new approach is that in light of errors the tempdirs will be in `/tmp` instead of `/etc/kubernetes/tmp`. Unsure if that's a blocker for this PR though.